Index: .idea/gradle.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/gradle.xml	(date 1517753368000)
+++ .idea/gradle.xml	(date 1519501554000)
@@ -11,11 +11,11 @@
             <option value="$PROJECT_DIR$/app" />
             <option value="$PROJECT_DIR$/cache" />
             <option value="$PROJECT_DIR$/domain" />
-            <option value="$PROJECT_DIR$/entities" />
             <option value="$PROJECT_DIR$/interactor" />
             <option value="$PROJECT_DIR$/repository" />
             <option value="$PROJECT_DIR$/storage" />
             <option value="$PROJECT_DIR$/tools" />
+            <option value="$PROJECT_DIR$/uimodel" />
           </set>
         </option>
         <option name="resolveModulePerSourceSet" value="false" />
Index: .idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/modules.xml	(date 1517753368000)
+++ .idea/modules.xml	(date 1519501554000)
@@ -6,11 +6,11 @@
       <module fileurl="file://$PROJECT_DIR$/app/app.iml" filepath="$PROJECT_DIR$/app/app.iml" />
       <module fileurl="file://$PROJECT_DIR$/cache/cache.iml" filepath="$PROJECT_DIR$/cache/cache.iml" />
       <module fileurl="file://$PROJECT_DIR$/domain/domain.iml" filepath="$PROJECT_DIR$/domain/domain.iml" />
-      <module fileurl="file://$PROJECT_DIR$/entities/entities.iml" filepath="$PROJECT_DIR$/entities/entities.iml" />
       <module fileurl="file://$PROJECT_DIR$/interactor/interactor.iml" filepath="$PROJECT_DIR$/interactor/interactor.iml" />
       <module fileurl="file://$PROJECT_DIR$/repository/repository.iml" filepath="$PROJECT_DIR$/repository/repository.iml" />
       <module fileurl="file://$PROJECT_DIR$/storage/storage.iml" filepath="$PROJECT_DIR$/storage/storage.iml" />
       <module fileurl="file://$PROJECT_DIR$/tools/tools.iml" filepath="$PROJECT_DIR$/tools/tools.iml" />
+      <module fileurl="file://$PROJECT_DIR$/uimodel/entities.iml" filepath="$PROJECT_DIR$/uimodel/entities.iml" />
     </modules>
   </component>
 </project>
\ No newline at end of file
Index: app/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/build.gradle	(date 1517753368000)
+++ app/build.gradle	(date 1519501554000)
@@ -1,7 +1,6 @@
 apply plugin: 'com.android.application'
-
 apply plugin: 'kotlin-android'
-
+apply plugin: 'kotlin-kapt'
 apply plugin: 'kotlin-android-extensions'
 
 android {
@@ -20,6 +19,11 @@
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
+        debug {
+            minifyEnabled true
+            useProguard false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
     }
     compileOptions {
         sourceCompatibility JavaVersion.VERSION_1_8
@@ -28,7 +32,7 @@
 }
 
 dependencies {
-    rootProject.ext.appDependencies.each {
+    rootProject.ext.uimodelDependencies.each {
         add(it.configuration, it.dependency, it.options)
     }
 }
Index: app/src/main/res/values/styles.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/styles.xml	(date 1517753368000)
+++ app/src/main/res/values/styles.xml	(date 1519501554000)
@@ -1,7 +1,7 @@
 <resources>
 
     <!-- Base application theme. -->
-    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
+    <style name="AppTheme" parent="@style/Theme.AppCompat.Light.NoActionBar">
         <!-- Customize your theme here. -->
         <item name="colorPrimary">@color/colorPrimary</item>
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
Index: cache/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- cache/build.gradle	(date 1517753368000)
+++ cache/build.gradle	(date 1519501554000)
@@ -1,7 +1,6 @@
 apply plugin: 'com.android.library'
-
 apply plugin: 'kotlin-android'
-
+apply plugin: 'kotlin-kapt'
 apply plugin: 'kotlin-android-extensions'
 
 android {
Index: dependencies.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- dependencies.gradle	(date 1517753368000)
+++ dependencies.gradle	(date 1519501554000)
@@ -24,7 +24,7 @@
             tools     : ":tools",
             storage   : ":storage",
             cache     : ":cache",
-            entities  : ":entities"
+            uimodel   : ":uimodel"
     ]
     def libs = [
             appCompat             : "com.android.support:appcompat-v7:${versionAndroidSDK}",
@@ -59,37 +59,42 @@
             androidTestCompile: "com.android.support.test.espresso:espresso-core:${versionEspressoCore}"
     ]
 
-    def compile = 'compile'
-    def api = 'api'
     def impl = 'implementation'
     def testImpl = 'testImplementation'
-    def testCompile = 'testCompile'
     def androidTestImpl = 'androidTestImplementation'
     def annotationProcessor = 'annotationProcessor'
-    def entities = 'entities'
 
     appDependencies = [
             [configuration: impl, dependency: project(modules.interactor)],
             [configuration: impl, dependency: project(modules.storage)],
+            [configuration: impl, dependency: project(modules.repository)],
             [configuration: impl, dependency: project(modules.cache)],
+            [configuration: impl, dependency: project(modules.domain)],
+            [configuration: impl, dependency: project(modules.tools)],
+            [configuration: impl, dependency: project(modules.uimodel)],
 
             [configuration: impl, dependency: libs.appCompat],
             [configuration: impl, dependency: libs.constraintLayout],
             [configuration: impl, dependency: libs.design],
+            [configuration: impl, dependency: libs.annotations],
 
             [configuration: impl, dependency: libs.archRuntime],
             [configuration: impl, dependency: libs.archExtensions],
             [configuration: annotationProcessor, dependency: libs.archCompiler],
 
+            [configuration: impl, dependency: libs.javaRx],
             [configuration: impl, dependency: libs.androidRx],
 
             [configuration: impl, dependency: libs.butterknife],
             [configuration: annotationProcessor, dependency: libs.butterknifeCompiler],
 
+            [configuration: impl, dependency: libs.dagger],
             [configuration: impl, dependency: libs.daggerAndroidSupport, options: { exclude module: 'jsr305' }],
             [configuration: annotationProcessor, dependency: libs.daggerCompiler],
             [configuration: annotationProcessor, dependency: libs.daggerAndroidProcessor],
 
+            [configuration: impl, dependency: libs.kotlin],
+
             [configuration: testImpl, dependency: testLibs.junit],
 
             // androidTestCompile
@@ -100,56 +105,79 @@
 
     interactorDependencies = [
             [configuration: impl, dependency: project(modules.repository)],
+            [configuration: impl, dependency: project(modules.domain)],
 
-            [configuration: testCompile, dependency: testLibs.junit]
+            [configuration: impl, dependency: libs.annotations],
+
+            [configuration: impl, dependency: libs.javaRx],
+            [configuration: impl, dependency: libs.dagger],
+
+            [configuration: impl, dependency: libs.kotlin],
+
+            [configuration: testImpl, dependency: testLibs.junit]
     ]
 
     repositoryDependencies = [
-            [configuration: compile, dependency: project(modules.domain)],
-            [configuration: compile, dependency: project(modules.tools)],
+            [configuration: impl, dependency: project(modules.domain)],
+            [configuration: impl, dependency: project(modules.tools)],
 
-            [configuration: compile, dependency: libs.javaRx],
-            [configuration: compile, dependency: libs.dagger],
+            [configuration: impl, dependency: libs.annotations],
 
-            [configuration: testCompile, dependency: testLibs.junit]
+            [configuration: impl, dependency: libs.javaRx],
+            [configuration: impl, dependency: libs.dagger],
+
+            [configuration: impl, dependency: libs.kotlin],
+
+            [configuration: testImpl, dependency: testLibs.junit]
     ]
 
     domainDependencies = [
-            [configuration: compile, dependency: libs.annotations],
+            [configuration: impl, dependency: libs.annotations],
 
-            [configuration: compile, dependency: libs.kotlin]
+            [configuration: impl, dependency: libs.kotlin]
     ]
 
-    entitiesDependencies = [
+    uimodelDependencies = [
             [configuration: impl, dependency: libs.firebaseDatabase],
 
             [configuration: impl, dependency: libs.kotlin]
     ]
 
     storageDependencies = [
-            [configuration: api, dependency: project(modules.domain)],
-            [configuration: api, dependency: project(modules.repository)],
-            [configuration: api, dependency: project(modules.entities)],
+            [configuration: impl, dependency: project(modules.domain)],
+            [configuration: impl, dependency: project(modules.repository)],
+            [configuration: impl, dependency: project(modules.uimodel)],
+            [configuration: impl, dependency: project(modules.tools)],
 
-            [configuration: api, dependency: libs.dagger],
+            [configuration: impl, dependency: libs.annotations],
+
+            [configuration: impl, dependency: libs.javaRx],
+            [configuration: impl, dependency: libs.dagger],
 
             //what does it mean to add a dependency with explicit version, so compile and runtime have both the same version
-            [configuration: api, dependency: libs.firebaseDatabase],
-            [configuration: api, dependency: libs.firebaseStorage],
-            [configuration: api, dependency: libs.firebaseCore]
+            [configuration: impl, dependency: libs.firebaseDatabase],
+            [configuration: impl, dependency: libs.firebaseStorage],
+            [configuration: impl, dependency: libs.firebaseCore],
+
+            [configuration: impl, dependency: libs.kotlin]
     ]
 
     cacheDependencies = [
             [configuration: impl, dependency: project(modules.repository)],
+            [configuration: impl, dependency: project(modules.domain)],
 
-            [configuration: impl, dependency: libs.javaRx]
+            [configuration: impl, dependency: libs.annotations],
+
+            [configuration: impl, dependency: libs.javaRx],
+
+            [configuration: impl, dependency: libs.kotlin]
     ]
 
     toolsDependencies = [
-            [configuration: compile, dependency: libs.annotations],
+            [configuration: impl, dependency: libs.annotations],
 
-            [configuration: compile, dependency: libs.javaRx],
+            [configuration: impl, dependency: libs.javaRx],
 
-            [configuration: compile, dependency: libs.kotlin]
+            [configuration: impl, dependency: libs.kotlin]
     ]
 }
Index: domain/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- domain/build.gradle	(date 1517753368000)
+++ domain/build.gradle	(date 1519501554000)
@@ -1,5 +1,6 @@
 apply plugin: 'java-library'
 apply plugin: 'kotlin'
+apply plugin: 'kotlin-kapt'
 
 dependencies {
     rootProject.ext.domainDependencies.each {
Index: interactor/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- interactor/build.gradle	(date 1517753368000)
+++ interactor/build.gradle	(date 1519501554000)
@@ -1,5 +1,6 @@
 apply plugin: 'java-library'
 apply plugin: 'kotlin'
+apply plugin: 'kotlin-kapt'
 
 dependencies {
     rootProject.ext.interactorDependencies.each {
Index: repository/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- repository/build.gradle	(date 1517753368000)
+++ repository/build.gradle	(date 1519501554000)
@@ -1,5 +1,6 @@
 apply plugin: 'java-library'
 apply plugin: 'kotlin'
+apply plugin: 'kotlin-kapt'
 
 dependencies {
     rootProject.ext.repositoryDependencies.each {
Index: settings.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- settings.gradle	(date 1517753368000)
+++ settings.gradle	(date 1519501554000)
@@ -1,1 +1,1 @@
-include ':app', ':domain', ':interactor', ':repository', ':tools', ':storage', ':cache', ':entities'
+include ':app', ':domain', ':interactor', ':repository', ':tools', ':storage', ':cache', ':uimodel'
Index: storage/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- storage/build.gradle	(date 1517753368000)
+++ storage/build.gradle	(date 1519501554000)
@@ -1,7 +1,6 @@
 apply plugin: 'com.android.library'
-
 apply plugin: 'kotlin-android'
-
+apply plugin: 'kotlin-kapt'
 apply plugin: 'kotlin-android-extensions'
 
 android {
Index: storage/src/main/java/com/admin/theapp/firebase/utils/mappers/FirebaseHotelModelToHotelMapper.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- storage/src/main/java/com/admin/theapp/firebase/utils/mappers/FirebaseHotelModelToHotelMapper.kt	(date 1517753368000)
+++ storage/src/main/java/com/admin/theapp/firebase/utils/mappers/FirebaseHotelModelToHotelMapper.kt	(date 1519501554000)
@@ -7,9 +7,7 @@
 import javax.inject.Singleton
 
 @Singleton
-class FirebaseHotelModelToHotelMapper @Inject
-internal constructor()// for dagger2
-    : Mapper<FirebaseHotelModel, Hotel>() {
+class FirebaseHotelModelToHotelMapper @Inject internal constructor() : Mapper<FirebaseHotelModel, Hotel>() {
 
     override fun map(firebaseHotelModel: FirebaseHotelModel): Hotel {
         return Hotel(
Index: tools/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- tools/build.gradle	(date 1517753368000)
+++ tools/build.gradle	(date 1519501554000)
@@ -1,5 +1,6 @@
 apply plugin: 'java-library'
 apply plugin: 'kotlin'
+apply plugin: 'kotlin-kapt'
 
 dependencies {
     rootProject.ext.toolsDependencies.each {
Index: entities/.gitignore
===================================================================
--- entities/.gitignore	(date 1517753368000)
+++ uimodel/.gitignore	(date 1519501554000)
@@ -1,0 +1,0 @@
Index: entities/build.gradle
===================================================================
--- entities/build.gradle	(date 1517753368000)
+++ uimodel/build.gradle	(date 1519501554000)
@@ -1,7 +1,6 @@
 apply plugin: 'com.android.library'
-
 apply plugin: 'kotlin-android'
-
+apply plugin: 'kotlin-kapt'
 apply plugin: 'kotlin-android-extensions'
 
 android {
@@ -27,7 +26,7 @@
 }
 
 dependencies {
-    rootProject.ext.entitiesDependencies.each {
+    rootProject.ext.uimodelDependencies.each {
         add(it.configuration, it.dependency, it.options)
     }
 }
\ No newline at end of file
Index: entities/proguard-rules.pro
===================================================================
--- entities/proguard-rules.pro	(date 1517753368000)
+++ uimodel/proguard-rules.pro	(date 1519501554000)
@@ -1,0 +1,0 @@
Index: entities/src/main/AndroidManifest.xml
===================================================================
--- entities/src/main/AndroidManifest.xml	(date 1517753368000)
+++ uimodel/src/main/AndroidManifest.xml	(date 1519501554000)
@@ -1,0 +1,0 @@
Index: entities/src/main/java/com/theapp/entities/FirebaseHotelModel.kt
===================================================================
--- entities/src/main/java/com/theapp/entities/FirebaseHotelModel.kt	(date 1517753368000)
+++ uimodel/src/main/java/com/theapp/entities/FirebaseHotelModel.kt	(date 1519501554000)
@@ -1,0 +1,0 @@
Index: entities/src/main/java/com/theapp/entities/HotelModel.kt
===================================================================
--- entities/src/main/java/com/theapp/entities/HotelModel.kt	(date 1517753368000)
+++ uimodel/src/main/java/com/theapp/entities/HotelModel.kt	(date 1519501554000)
@@ -1,0 +1,0 @@
Index: entities/src/main/res/values/strings.xml
===================================================================
--- entities/src/main/res/values/strings.xml	(date 1517753368000)
+++ uimodel/src/main/res/values/strings.xml	(date 1519501554000)
@@ -1,0 +1,0 @@
Index: domain/build/tmp/jar/MANIFEST.MF
===================================================================
--- domain/build/tmp/jar/MANIFEST.MF	(date 1517753368000)
+++ domain/build/tmp/jar/MANIFEST.MF	(date 1517753368000)
@@ -1,2 +0,0 @@
-Manifest-Version: 1.0
-
